ui/plugin-system/src/runtime/DataQueriesProvider/model.ts

```diff
@@ -60,26 +60,37 @@ export function transformQueryResults(results: UseQueryResult[], definitions: Qu
 
 export function useQueryType(): (pluginKind: string) => string | undefined {
   const { data: timeSeriesQueryPlugins, isLoading } = useListPluginMetadata('TimeSeriesQuery');
+  const { data: traceQueryPlugins, isLoading: isTraceQueryPluginLoading } = useListPluginMetadata('TraceQuery');
 
+  // For example, `map: {"TimeSeriesQuery":["PrometheusTimeSeriesQuery"],"TraceQuery":["TempoTraceQuery"]}`
   const queryTypeMap = useMemo(() => {
     const map: Record<string, string[]> = {
       TimeSeriesQuery: [],
+      TraceQuery: [],
     };
 
     if (timeSeriesQueryPlugins) {
       timeSeriesQueryPlugins.forEach((plugin) => {
-        map['TimeSeriesQuery']?.push(plugin.kind);
+        map[plugin.pluginType]?.push(plugin.kind);
       });
     }
 
+    if (traceQueryPlugins) {
+      traceQueryPlugins.forEach((plugin) => {
+        map[plugin.pluginType]?.push(plugin.kind);
+      });
+    }
     return map;
-  }, [timeSeriesQueryPlugins]);
+  }, [timeSeriesQueryPlugins, traceQueryPlugins]);
 
   const getQueryType = useCallback(
     (pluginKind: string) => {
       if (isLoading) {
         return undefined;
       }
+      if (isTraceQueryPluginLoading) {
```

Comment:
Don't you need to use there one or the other "isLoading" depending on the pluginKind ?